File: DFS-frontend/modules/dfsmodule/android/src/main/java/expo/modules/dfsmodule/DfsmoduleModuleSharedObject.kt
package expo.modules.dfsmodule

import expo.modules.kotlin.AppContext
import expo.modules.kotlin.sharedobjects.SharedObject

class DfsmoduleModuleSharedObject(appContext: AppContext) : SharedObject(appContext) {
  var count: Int = 0

  override fun sharedObjectDidRelease() {
    super.sharedObjectDidRelease()
  }
}

File: DFS-frontend/modules/dfsmodule/src/DfsmoduleModuleSharedObject.ts
import { SharedObject, useReleasingSharedObject } from 'expo-modules-core';

import DfsmoduleModule from './DfsmoduleModule';

export declare class DfsmoduleModuleSharedObject extends SharedObject {
  count: number;
}

/**
 * Creates a new DfsmoduleModuleSharedObject instance.
 * You are responsible for releasing it from memory by calling `release()` when done.
 */
export function createDfsmoduleModuleSharedObject(): DfsmoduleModuleSharedObject {
  return new DfsmoduleModule.DfsmoduleModuleSharedObject();
}

/**
 * A hook that creates a DfsmoduleModuleSharedObject instance and automatically
 * releases it when the component unmounts.
 */
export function useDfsmoduleModuleSharedObject(): DfsmoduleModuleSharedObject {
  return useReleasingSharedObject(() => new DfsmoduleModule.DfsmoduleModuleSharedObject(), []);
}
